dashboard/app/models/user.rb

```diff
@@ -1696,6 +1698,10 @@ def self.track_level_progress(
         user_level.locale_supported = script.supported_locale?(locale)
       end
 
+      if unit_group
+        user_level.unit_group_id = unit_group.id
```

Comment:
it looks like this will update the unit group id if one exists (the `first` case in `first_or_initialize`), whereas the PR description says it will only set unit group id when creating user levels. I think either could be ok, but it seems important to be consistent about whether we do it on update in addition to create, and also the field should be documented with a comment to be clear about whether it reflects the initial value or the last update.  this could be one place to document it: https://github.com/code-dot-org/code-dot-org/blob/195accb47046bfa0e16e3a847628670afe737700/dashboard/app/models/user_level.rb#L29-L31

Reply:
@daynew due to the larger number of codepaths where we update the user level, I think it will be best to only set this when creating the user level. it's possible that `user_level.persisted?` will tell you whether it is a create or an update.